applications/rapid-analysis/templates/comcam-gather2aAos-set.yaml

```diff
@@ -0,0 +1,249 @@
+            - name: {{ $env_var }}
+              value: {{ $env_value | quote }}
+          {{- end }}
+          {{- range $env := $.Values.envSecrets }}
+            - name: {{ $env.name }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $env.secretName }}
+                  key: {{ $env.secretKey }}
+          {{- end }}
+          {{- end }}
+          {{- if $.Values.butlerSecret }}
+            - name: PGPASSFILE
+              value: "{{ $.Values.butlerSecret.containerPath }}/postgres-credentials.txt"
+            - name: PGUSER
+              value: {{ $.Values.butlerSecret.dbUser | quote }}
+            - name: AWS_SHARED_CREDENTIALS_FILE
+              value: "{{ $.Values.butlerSecret.containerPath }}/aws-credentials.ini"
+          {{- end }}
+          {{- if $.Values.redis.enabled }}
+            - name: REDIS_HOST
+              value: "redis-service"
+          {{- if $.Values.redis.envSecrets }}
+          {{- range $env := $.Values.redis.envSecrets }}
+            - name: {{ $env.name }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $env.secretName }}
+                  key: {{ $env.secretKey }}
+          {{- end }}
+          {{- end }}
+          {{- end }}
+          envFrom:
+            - configMapRef:
+                name: env-configmap
+          volumeMounts:
+            - name: rubintv-creds
+              mountPath: "/etc/rubintv/creds"
+              readOnly: true
+          {{- if $.Values.nfsMountpoint }}
+          {{- range $values := $.Values.nfsMountpoint }}
+          {{- if (has $values.name $script.mounts) }}
+            - name: {{ $values.name }}
+              mountPath: {{ $values.containerPath }}
+              readOnly: {{ $values.readOnly }}
+          {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- if $.Values.pvcMountpoint }}
+          {{- range $values := $.Values.pvcMountpoint }}
+          {{- if (has $values.name $script.mounts) }}
+            - name: {{ $values.name }}
+              mountPath: {{ $values.containerPath }}
+              {{- if ($values.subPath) }}
+              subPath: {{ $values.subPath }}
+              {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- if $.Values.pvcMountpointClaim }}
+          {{- range $values := $.Values.pvcMountpointClaim }}
+          {{- if (has $values.name $script.mounts) }}
+            - name: {{ $values.name }}
+              mountPath: {{ $values.containerPath }}
+              {{- if ($values.subPath) }}
+              subPath: {{ $values.subPath }}
+              {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- end }}
+          {{- if $.Values.butlerSecret }}
+            - name: {{ $.Release.Name }}-butler-secret
+              mountPath: {{ $.Values.butlerSecret.containerPath }}
+          {{- end }}
+          {{- if or $.Values.resources $script.resources }}
```

Comment:
There are a bunch of these ".values or script" blocks (nodeSelector, affinity, tolerations, etc), which I think can each be replaced by something like:
```
{{- if or $.Values.resources $script.resources }}
{{- $resources := $script.resources | default $.Values.resources }}
resources:
  {{- toYaml $resources | nindent 12 }}
{{- end }}
```
I've tried this out and it seems to behave correctly for all the combinations I could come up with. Or you could even inline it as `{{- $script.resources | default $.Values.resources | toYaml | nindent 12}}`, but I haven't tested that one. Still requires the if statement as a guard around the resources block.